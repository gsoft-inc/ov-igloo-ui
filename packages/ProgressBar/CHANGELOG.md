# @igloo-ui/progress-bar

## 0.1.0

### Minor Changes

- 74ce3e3: initial release of ProgressBar
