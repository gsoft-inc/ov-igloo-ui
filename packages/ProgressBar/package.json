{
  "name": "@igloo-ui/progress-bar",
  "version": "0.1.1",
  "main": "dist/ProgressBar.js",
  "module": "dist/ProgressBar.js",
  "types": "dist/ProgressBar.d.ts",
  "styles": "dist/progress-bar.css",
  "publishConfig": {
    "access": "public"
  },
  "repository": "git+https://github.com/gsoft-inc/ov-igloo-ui.git",
  "author": "Officevibe Inc.",
  "contributors": [
    "Franck franck.gaudin@gsoft.com"
  ],
  "license": "Apache-2.0",
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c rollup.config.js"
  },
  "dependencies": {
    "@igloo-ui/tokens": "^2.0.0",
    "classnames": "^2.3.2"
  },
  "devDependencies": {
    "@igloo-ui/helper-text": "^0.2.8"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "IE >= 11"
  ]
}
