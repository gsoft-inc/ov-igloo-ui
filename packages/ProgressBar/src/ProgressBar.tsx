import * as React from "react";
import cx from "classnames";

import "./progress-bar.scss";

export interface ProgressBarProps extends React.ComponentProps<"div"> {
    /** Sets the value of the progress bar, between 0 and 1 inclusive. */
    value: number;
    /** True for a compact appearance. */
    isCompact?: boolean;
    /** Add a specific class to the component. */
    className?: string;
    /** Add a data-test tag for automated tests. */
    dataTest?: string;
    /** Defines a string value that labels the current element. */
    ariaLabel?: string;
}

const ProgressBar: React.FunctionComponent<ProgressBarProps> = ({
    value,
    isCompact,
    className,
    dataTest,
    ariaLabel,
    ...rest
}: ProgressBarProps) => {
    let progress = value * 100;
    if (value > 1) {
        progress = 100;
    }

    const isCompleted = progress === 100;

    const classes = cx("ids-progress-bar", className, {
        "ids-progress-bar--completed": isCompleted,
        "ids-progress-bar--compact": isCompact
    });

    return (
        <div
            role="progressbar"
            aria-label={ariaLabel}
            className={classes}
            data-test={dataTest}
            {...rest}
        >
            <span
                className="ids-progress-bar__status"
                style={{ width: `${progress}%` }}
            />
        </div>
    );
};

export default ProgressBar;
