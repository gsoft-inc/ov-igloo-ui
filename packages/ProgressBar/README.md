# ProgressBar

Progress bar that displays linear progress. Works best with elements that do not involve specific steps.

<Example is="custom" />

<ReferenceLinks is="custom" />

## Installation

To install `@igloo-ui/progress-bar` in your project, you will need to run the following command using [npm](https://www.npmjs.com/):

```bash
npm install @igloo-ui/progress-bar
```

If you prefer [Yarn](https://classic.yarnpkg.com/en/), use the following command instead:

```bash
yarn add @igloo-ui/progress-bar
```

## Usage

Then to use the component in your code just import it!

```jsx
import ProgressBar from '@igloo-ui/progress-bar';

<ProgressBar value={0.7} ariaLabel="Progress: 70% of objectives achieved" />;
```
