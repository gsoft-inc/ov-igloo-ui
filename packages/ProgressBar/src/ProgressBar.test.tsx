/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow } from 'enzyme';

import ProgressBar from './ProgressBar';

describe('ProgressBar', () => {
  const component = shallow(<ProgressBar value={0.7} />);
  test('It should render without errors', () => {
    const wrapper = component.find('.ids-progress-bar');
    expect(wrapper.length).toBe(1);
  });

  test('It should render a snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
