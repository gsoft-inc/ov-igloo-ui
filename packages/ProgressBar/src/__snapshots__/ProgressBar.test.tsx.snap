// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar It should render a snapshot 1`] = `ShallowWrapper {}`;
